xarray/test/test_dataset.py

```diff
@@ -1849,6 +1849,141 @@ def test_convert_dataframe_with_many_types_and_multiindex(self):
 
+    def test_to_and_from_dict(self):
+        # <xarray.Dataset>
+        # Dimensions:  (t: 10)
+        # Coordinates:
+        #   * t        (t) <U1 'a' 'b' 'c' 'd' 'e' 'f' 'g' 'h' 'i' 'j'
+        # Data variables:
+        #     a        (t) float64 0.6916 -1.056 -1.163 0.9792 -0.7865 0.7831 0.4174 ...
+        #     b        (t) float64 1.32 0.1954 1.91 1.39 0.519 -0.2772 -1.316 1.111 ...
+        x = np.random.randn(10)
+        y = np.random.randn(10)
+        t = list('abcdefghij')
+        ds = Dataset(OrderedDict([('a', ('t', x)),
+                                  ('b', ('t', y)),
+                                  ('t', ('t', t))]))
+        expected = {
+                    'coords': {
+                               't':{
+                                    'dims': ('t',),
+                                    'data': t,
+                                    'attrs': {}
+                                    }
+                              },
+                    'attrs': {}, 
+                    'dims': {'t': 10},
+                    'data_vars': {
+                                  'a':{
+                                       'dims': ('t',),
+                                       'data': x.tolist(),
+                                       'attrs': {}
+                                       },
+                                  'b':{
+                                       'dims': ('t',),
+                                       'data': y.tolist(),
+                                       'attrs': {}
+                                       }   
+                                  }
+                   }
+
+        actual = ds.to_dict()
+
+        # check that they are identical
+        self.assertEqual(expected, actual)
+
+        # check roundtrip
+        self.assertDatasetIdentical(ds, Dataset.from_dict(actual))
+
+        # verify coords are included roundtrip
+        expected = ds.set_coords('b')
+        actual = Dataset.from_dict(expected.to_dict())
+
+        self.assertDatasetIdentical(expected, actual)
+
+        # test some incomplete dicts:
+        # this one has no attrs field, the dims are strings, and x, y are np.arrays
+
+        d = {
+            'coords': {
+                       't':{
+                            'dims': 't',
+                            'data': t
+                            }
+                      },
+            'dims': 't',
+            'data_vars': {
+                          'a':{
+                               'dims': 't',
+                               'data': x
+                               },
+                          'b':{
+                               'dims': 't',
+                               'data': y
+                               }   
+                          }
+           }
+        self.assertDatasetIdentical(ds, Dataset.from_dict(d))
+
+        # this is kind of a flattened version with no coords, or data_vars
+        d = {'a':{
```

Comment:
You have a few PEP8 style violations -- try running an automated tool like pep8 to catch these: https://pypi.python.org/pypi/pep8